### PR TITLE
Update `syn`, `quote` and `proc-macro2`

### PR DESCRIPTION
**Pull request by hcpl**

### CODE REVIEW

File: futures-macro-async/src/lib.rs [hunk 1]
```diff
@@ -573,19 +573,19 @@ if_nightly! {
         let mut spans = Tokens::new();
         tokens.to_tokens(&mut spans);
         let good_tokens = proc_macro2::TokenStream::from(spans).into_iter().collect::<Vec<_>>();
-        let first_span = good_tokens.first().map(|t| t.span).unwrap_or(Span::def_site());
-        let last_span = good_tokens.last().map(|t| t.span).unwrap_or(first_span);
+        let first_span = good_tokens.first().map(|t| t.span()).unwrap_or(Span::call_site());
```
Comment (hcpl) [inline futures-macro-async/src/lib.rs]: Should this really be `call_site` or better to make `first_last` function to return `proc_macro::Span`?

**Comment by withoutboats** [comment]
This looks good, just had github rebase it and will merge if its green :)

**Comment by withoutboats** [comment]
It looks like nightly has changed a bit more in the past week and there's a bit more needed to get this green.

**Comment by withoutboats** [comment]
Thanks! Looks good :heart:

**Pull request merged**